scripts/chk_shellscripts/chk_shellscripts.sh

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env bash
+
+set -e
+
+REPO_ROOT="$(dirname "$0")"/../..
+REPO_ROOT=$(realpath "${REPO_ROOT}")
+BASELINE_FILENAME="baseline.checksums"
+BASELINE_FILE="${REPO_ROOT}/scripts/chk_shellscripts/${BASELINE_FILENAME}"
+
+for arg in "$@"; do
+  case "$arg" in
+  --create-baseline) CREATE_BASELINE=1 ;;
+  *)
+    echo "Usage:"
+    echo "  $(basename "${0}") [--create-baseline]"
+    echo "  --create-baseline    create a baseline of current shell scripts. all errors"
```

Comment:
Do the tests take that long? This is not really the way we run tests....

Reply:
You mean the baseline stuff? This is not about reducing test time. It is about that we dont need to update all shell-scripts to conform with `shellcheck` right now. The baseline function is used to capture the hashes of the scripts and is used to determine if they changed. If unchanged, the script will be ignored and not checked with `shellsheck`. But if someone did a change to a script, because the hash changed, it is also needed to fix all the `shellcheck` related errors. Of course it is also possible to just set the new hash in the `baseline.checksums`, but this will be visible during review. However, new shell-scripts will always be checked with `shellcheck`. This is just about giving us some time to fix the shell scripts we currently have. So we can fix the scripts iteratively. We can have the `shellcheck` stuff in place, without the need to update at first all our shell-scripts.

Reply:
I see this being very annoying to use in practice, especially to people who do not know how this mechanism works.

We get the same benefit by using a blacklist without hashes, won't we?

Reply:
basically it will be described, if an error occurs.. the problem with blacklist may be that the scripts will just stay blacklisted and nobody will care about fixing these scripts.

Reply:
What is the conclusion? Do we go with this PR or not?

Reply:
I'm making it "to discuss"